### PR TITLE
Support VoiceOver

### PR DESCRIPTION
**Pull request by scannillo**
DTBTSDK-3248

### Reason for changes

- Our buttons did not properly support VoiceOver, a key accessibility feature for users

### Summary of changes

- Add [`accessibilityLabel`](https://developer.apple.com/documentation/objectivec/nsobject/1615181-accessibilitylabel)s to each button, so VoiceOver properly reads a full description of the button text

### Videos

| Button Type      | Video Link |
| ----------- | ----------- |
| PayPal Credit      |   https://github.com/paypal/paypal-ios/assets/35243507/89b43a90-948f-4411-95c8-246a06d6f389 |
| PayPal Pay Later   | https://github.com/paypal/paypal-ios/assets/35243507/85922bb2-deac-476f-9799-3130b77de2ee |
| PayPal w/ Suffix Text  |    https://github.com/paypal/paypal-ios/assets/35243507/2c32ea00-aeaa-4cf2-87c7-bfca700463d7 |
| PayPal w/ Prefix Text  |    https://github.com/paypal/paypal-ios/assets/35243507/e6974bd7-d6bd-4d14-b080-7f17bb895631 |
| Small PayPal Logo  |    https://github.com/paypal/paypal-ios/assets/35243507/d158e99e-65f1-4325-b149-746ab94efc95  |

### Checklist

- [X] Added a changelog entry
- [ ] On device VoiceOver check

### Authors
@scannillo

### CODE REVIEW

**Review by stechiu — APPROVED** [review]
Tested in accessibility inspector (couldn't get the demo to run on device) and everything looks good.

**Comment by scannillo** [comment]
> couldn't get the demo to run on device

Darn, I made a ticket for us to address this (DTPPCPSDK-1696). I think it might have stemmed from [this PR](https://github.com/paypal/paypal-ios/pull/209).

File: Sources/PaymentButtons/ImageAsset.swift [hunk 1]
```diff
@@ -28,7 +47,15 @@ enum ImageAsset {
...
+        return imageAssetString
+    }
+    
+    private let button: PaymentButton
+    
+    // MARK: - Initializer
+        
+    init(button: PaymentButton) {
+        self.button = button
     }
```
Comment (sshropshire) [inline Sources/PaymentButtons/ImageAsset.swift]:
```
init(fundingSource: FundingSource) {
    self.fundingSource = fundingSource
}
```
nit: take or leave we could pass the funding source and omit the button reference here.

**Review by sshropshire — APPROVED** [review]

File: Sources/PaymentButtons/ImageAsset.swift (hunk 1 above)
Comment (scannillo) [reply Sources/PaymentButtons/ImageAsset.swift]: Good eye - so we also reference the button.color ([code](https://github.com/paypal/paypal-ios/blob/73e3e0c5f80f0019254e1c3ac9ba3b755d7c4729/Sources/PaymentButtons/ImageAsset.swift#L40)) & button.size ([code](https://github.com/paypal/paypal-ios/blob/73e3e0c5f80f0019254e1c3ac9ba3b755d7c4729/Sources/PaymentButtons/ImageAsset.swift#L29)), in addition to the fundingSource.

The current setup doesn't feel like the best design to me. An extension on PaymentButton probably makes more sense than a separate class, or breaking up PaymentButton to separate responsibilities out more.

Comment (scannillo) [reply Sources/PaymentButtons/ImageAsset.swift]: I went with an extension (see commit 957ffdcb7f7bab36bad96ecf1491875cdf60aa40) for now. Let me know what you think!

Comment (sshropshire) [reply Sources/PaymentButtons/ImageAsset.swift]: 👏 🚢

File: Sources/PaymentButtons/PaymentButton+ImageAsset.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        UIImage(named: fileName, in: PaymentButton.bundle, compatibleWith: nil)
+    }
+    
+    var imageAccessibilityLabel: String {
+        // NEXT_MAJOR_VERSION: - To be replaced with translation strings.
+        if fileName.starts(with: "credit") {
+            return "PayPal Credit"
+        } else {
+            return "PayPal"
+        }
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PaymentButton+ImageAsset.swift]: Take it or leave it but we could clean this up:
```suggestion
        fileName.starts(with: "credit") ? "PayPal Credit" : "PayPal"
```

Comment (jaxdesmarais) [inline Sources/PaymentButtons/PaymentButton+ImageAsset.swift]: Take it or leave it but we could clean this up:
```suggestion
        fileName.starts(with: "credit") ? "PayPal Credit" : "PayPal"
```

File: Sources/PaymentButtons/PaymentButton+ImageAsset.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            imageAssetString += "color"
+
+        case .black, .darkBlue:
+            imageAssetString += "monochrome"
+
+        case .blue:
+            imageAssetString += "blue"
+        }
+        
+        return imageAssetString
```
Comment (jaxdesmarais) [inline Sources/PaymentButtons/PaymentButton+ImageAsset.swift:47]: Can 🏈 on this but is it worth doing a check for an empty `imageAssetString` anywhere? Is there ever an instance where we'd fall through all of these switches and end up with the empty string returned from line 24?

Comment (scannillo) [reply Sources/PaymentButtons/PaymentButton+ImageAsset.swift:47]: Yeh, agreed. This empty string default isn't a good practice and not very error-proof. I know Stephanie's PR here https://github.com/paypal/paypal-ios/pull/235/files updated these to be full string names that match the asset files, versus concatenating them - which is an improvement.

But we should def further clean this up. Gonna leave it for a separate PR 🏈

**Review by jaxdesmarais — APPROVED** [review]

**Review by richherrera — APPROVED** [review]

**Pull request merged**